csrc/tensor_metadata.cpp

```diff
@@ -209,65 +209,66 @@ class BackwardTraverseFromLogicalToAlloc {
 };
 
 void validateAllocationSizesAndStrides(
-    const std::vector<IterDomain*>& alloc_dom_no_reductions,
+    const std::vector<IterDomain*>& alloc_dom,
```

Comment:
This is mostly a clean-up. I may have fixed a few bugs and/or hardened the validation, but I don't quite remember what was broken exactly...